code/game/objects/items/weapons/weaponry.dm

```diff
@@ -21,7 +25,13 @@
 		M << "You name the weapon [input]. Say hello to your new friend."
 		return
 
+/obj/item/weapon/proc/redesc_weapon(mob/M)
+	var/input = stripped_input(M,"Describe your object here", ,"", 100)
 
+	if(src && input && !M.stat && in_range(M,src) && !M.restrained() && M.canmove)

```

Comment:
if(!qdeleted(src) && canUseTopic(src, be_close = TRUE))

   update rename_weapon to use the same check if does not already